matchms/importing/load_from_msp.py

```diff
@@ -149,16 +147,20 @@ def parse_metadata(rline: str, params: dict):
     matches = []
     splitted_line = rline.split(":", 1)
     if splitted_line[0].lower() == 'comments' and "=" in splitted_line[1]:
-        pattern = r'(\S+)="([^"]*)"|"(\w+)=([^"]*)"|"([^"]*)=([^"]*)"|(\S+)=(\d+(?:\.\d*)?)'
+        pattern = r'(\S+)="([^"]+)"|' \
+                  r'"(\w+)=([^"]+)"|' \
+                  r'"([^"]+)=([^"]+)"|' \
+                  r'(\S+)=(\d+(?:\.\d*)?)'
         matches = re.findall(pattern, splitted_line[1].replace("'", '"'))
         for match in matches:
             match = [i for i in match if i]
-            key = match[0]
-            value = match[1]
-            if key.lower().strip() in params.keys() and key.lower().strip() == 'smiles':
-                params[key.lower()+"_2"] = value.strip()
-            else:
-                params[key.lower().strip()] = value.strip()
+            if len(match) == 2:
+                key = match[0]
+                value = match[1]
+                if key.lower().strip() in params.keys() and key.lower().strip() == 'smiles':
+                    params[key.lower()+"_2"] = value.strip()
+                else:
+                    params[key.lower().strip()] = value.strip()
     if len(matches) == 0:
         params[splitted_line[0].lower()] = splitted_line[1].strip()
 
```

Comment:
I think the len(matches) == 0 part, ensures that lines that contain something of the form compound:some name, is stored. This seems weir to me. But it might be something we want. To me it would make sense to remove this. What do you think?